packages/kbn-alerts-ui-shared/src/rule_form/rule_definition/rule_definition.test.tsx

```diff
@@ -60,160 +66,164 @@ const ruleModel: RuleTypeModel = {
+        id: 'test-id',
+        params: {},
+        schedule: {
+          interval: '1m',
+        },
+        alertDelay: {
+          active: 5,
+        },
+        notifyWhen: null,
+        consumer: 'stackAlerts',
+      },
+      selectedRuleType: ruleType,
+      selectedRuleTypeModel: ruleModel,
+    });
+
+    render(<RuleDefinition />);
 
     expect(screen.queryByTestId('ruleConsumerSelection')).not.toBeInTheDocument();
   });
 
   test('Can toggle advanced options', async () => {
-    render(
-      <RuleDefinition
-        requiredPlugins={requiredPlugins}
-        formValues={{
-          id: 'test-id',
-          params: {},
-          schedule: {
-            interval: '1m',
-          },
-          alertDelay: {
-            active: 5,
-          },
-          notifyWhen: null,
-          consumer: 'stackAlerts',
-        }}
-        selectedRuleType={ruleType}
-        selectedRuleTypeModel={ruleModel}
-        authorizedConsumers={['logs', 'stackAlerts']}
-        onChange={mockOnChange}
-      />
-    );
+    useRuleFormState.mockReturnValue({
+      plugins,
+      formData: {
+        id: 'test-id',
+        params: {},
+        schedule: {
+          interval: '1m',
+        },
+        alertDelay: {
+          active: 5,
+        },
+        notifyWhen: null,
+        consumer: 'stackAlerts',
+      },
+      selectedRuleType: ruleType,
+      selectedRuleTypeModel: ruleModel,
+    });
+
+    render(<RuleDefinition />);
 
     fireEvent.click(screen.getByTestId('advancedOptionsAccordionButton'));
-    expect(screen.getByText(ALERT_DELAY_TITLE)).toBeVisible();
+    expect(screen.getByTestId('alertDelay')).toBeVisible();
   });
 
   test('Calls onChange when inputs are modified', () => {
-    render(
-      <RuleDefinition
-        requiredPlugins={requiredPlugins}
-        formValues={{
-          id: 'test-id',
-          params: {},
-          schedule: {
-            interval: '1m',
-          },
-          alertDelay: {
-            active: 5,
-          },
-          notifyWhen: null,
-          consumer: 'stackAlerts',
-        }}
-        selectedRuleType={ruleType}
-        selectedRuleTypeModel={ruleModel}
-        authorizedConsumers={['logs', 'stackAlerts']}
-        onChange={mockOnChange}
-      />
-    );
+    useRuleFormState.mockReturnValue({
+      plugins,
+      formData: {
+        id: 'test-id',
+        params: {},
+        schedule: {
+          interval: '1m',
+        },
+        alertDelay: {
+          active: 5,
+        },
+        notifyWhen: null,
+        consumer: 'stackAlerts',
+      },
+      selectedRuleType: ruleType,
+      selectedRuleTypeModel: ruleModel,
+    });
+
+    render(<RuleDefinition />);
 
     fireEvent.change(screen.getByTestId('ruleScheduleNumberInput'), {
       target: {
         value: '10',
       },
     });
-    expect(mockOnChange).toHaveBeenCalledWith('interval', '10m');
+    expect(mockOnChange).toHaveBeenCalledWith({
+      type: 'setSchedule',
```

Comment:
All of the dispatch are related to the rule type params component, which is hard to test without mocking it. I think it would be better to test this with E2E tests.